server/project.go

```diff
@@ -247,61 +247,76 @@ func (s *Server) registerProjectRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusNotFound, fmt.Sprintf("VCS not found with ID: %d", repositoryCreate.VCSID))
 		}
 
-		repositoryCreate.WebhookURLHost = s.profile.ExternalURL
-		repositoryCreate.WebhookEndpointID = uuid.New().String()
-		secretToken, err := common.RandomString(gitlab.SecretTokenLength)
+		// We want to use only one webhook for Bytebase per repo/project.
+		repositories, err := s.store.FindRepository(ctx, &api.RepositoryFind{
+			WebURL: &repositoryCreate.WebURL,
+		})
 		if err != nil {
-			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to generate random secret token for GitLab").SetInternal(err)
-		}
-		repositoryCreate.WebhookSecretToken = secretToken
-
-		// Create a new webhook and retrieve the created webhook ID
-		var webhookCreatePayload []byte
-		switch vcs.Type {
-		case vcsPlugin.GitLabSelfHost:
-			webhookCreate := gitlab.WebhookCreate{
-				URL:                   fmt.Sprintf("%s/%s/%s", s.profile.ExternalURL, gitlabWebhookPath, repositoryCreate.WebhookEndpointID),
-				SecretToken:           repositoryCreate.WebhookSecretToken,
-				PushEvents:            true,
-				EnableSSLVerification: false, // TODO(tianzhou): This is set to false, be lax to not enable_ssl_verification
-			}
-			webhookCreatePayload, err = json.Marshal(webhookCreate)
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to find repository with web url: %s", repositoryCreate.WebURL)).SetInternal(err)
+		}
+
+		repositoryCreate.WebhookURLHost = s.profile.ExternalURL
+		// If we can find at least one repository with same web url, we will use the same webhook instead of create a new one.
+		if len(repositories) == 0 {
```

Comment:
Probably not related to this change but we may want to minimize the function size to make it more readable, e.g. moving the VCS resource creation logic to a separate function.

Reply:
Done.